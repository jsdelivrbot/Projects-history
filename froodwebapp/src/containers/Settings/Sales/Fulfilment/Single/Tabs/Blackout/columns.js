import React from 'react';
import { TableActionButtons } from 'components';

export default (handleDeleteRow, readonly) => {
  const columns = [{
    title: 'From',
    dataIndex: 'startDate',
    render: text => text
  }, {
    title: 'To',
    dataIndex: 'endDate',
    render: text => text
  }, {
    title: 'Name',
    dataIndex: 'name',
    render: text => text
  }];
  if (!readonly) {
    columns.push({
      title: 'Actions',
      dataIndex: 'total',
      render: (text, record) => (
        <TableActionButtons
          record={ record }
          editButtonVisible={ false }
          deleteButtonVisible
          handleDeactivate={ handleDeleteRow }
          deleteButtonDisabled={ readonly }
        />
      )
    });
    return columns;
  }
  return columns;
};
