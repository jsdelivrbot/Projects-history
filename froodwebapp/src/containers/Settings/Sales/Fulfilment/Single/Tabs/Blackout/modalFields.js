export default () => ([{
  header: 'From',
  type: 'datePicker',
  name: 'startDate',
  editableForExistingItem: true,
}, {
  header: 'To',
  type: 'datePicker',
  name: 'endDate',
  editableForExistingItem: true,
}, {
  header: 'Name',
  type: 'text',
  name: 'name',
  editableForExistingItem: true,
}]);
