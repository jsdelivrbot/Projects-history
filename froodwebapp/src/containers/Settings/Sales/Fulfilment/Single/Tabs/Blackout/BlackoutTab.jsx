import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row } from 'react-flexbox-grid';
import { connect } from 'react-redux';
import { Table, Spin } from 'antd';
import moment from 'moment';
import { ConnectedTopFormModal, NewTableRow } from 'components';
import {
  holidaysGetRequest,
  holidaySaveRequest,
  holidayDeleteRequest
} from 'redux-base/actions';
import { table } from 'styles/common.scss';
import columns from './columns';
import fields from './modalFields';

const mapStateToProps = state => ({
  holidays: state.transporters.holidays,
  needReloadHolidays: state.transporters.needReloadHolidays,
  loadingPage: state.transporters.loadingPage
});

const mapDispatchToProps = {
  holidaysGetRequest,
  holidaySaveRequest,
  holidayDeleteRequest
};

const newHoliday = {
  startDate: moment(),
  endDate: moment(),
  name: ''
};

export class BlackoutTab extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      modalData: newHoliday,
    };
  }

  componentWillMount() {
    this.props.holidaysGetRequest({
      id: this.props.fulfilmentId
    });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.needReloadHolidays) {
      this.setState({
        modalVisible: false,
        modalData: undefined
      }, () => this.props.holidaysGetRequest({
        id: this.props.fulfilmentId
      }));
    }
  }

  handleToggleModal = () => {
    this.setState({
      modalVisible: !this.state.modalVisible,
      modalData: newHoliday,
    });
  }

  handleSave = (holiday) => {
    const hol = holiday;
    // if we don`t change date on modal, it give us moment object, that`s why we check for this here
    hol.startDate = (typeof hol.startDate === 'object') ? hol.startDate.format('D-MMMM-YYYY') : hol.startDate;
    hol.endDate = (typeof hol.endDate === 'object') ? hol.endDate.format('D-MMMM-YYYY') : hol.endDate;
    this.setState({
      modalVisible: !this.state.modalVisible,
      modalData: newHoliday
    }, () => this.props.holidaySaveRequest({
      id: this.props.fulfilmentId,
      payload: hol
    }));
  }

  handleDeleteRow = (e) => {
    this.props.holidayDeleteRequest({
      id: this.props.fulfilmentId,
      holidayId: e.target.id
    });
  }

  handleAddRow = () => {
    this.setState({
      modalVisible: true,
      modalData: newHoliday
    });
  }

  render() {
    const {
      holidays,
      readonly,
      loadingPage
    } = this.props;

    const {
      modalData,
      modalVisible
    } = this.state;

    return (
      <div>
        <ConnectedTopFormModal
          loading={ loadingPage }
          title="New Holiday"
          handleSave={ this.handleSave }
          handleToggleModal={ this.handleToggleModal }
          visible={ modalVisible }
          initialValues={ modalData }
          fields={ fields() }
          okText="Add"
          buttonVisible={ false }
        />
        <Spin spinning={ loadingPage }>
          <Row>
            <Table
              rowKey="id"
              dataSource={ holidays }
              columns={ columns(this.handleDeleteRow, readonly) }
              size="small"
              pagination={ false }
              className={ table }
            />
            { !readonly && <NewTableRow
              addOtherRowText="Add Another Holiday"
              addNewRowText="Add New Holiday"
              hasData={ holidays && holidays.length !== 0 }
              onClick={ this.handleAddRow }
            /> }
          </Row>
        </Spin>
      </div>
    );
  }
}

BlackoutTab.propTypes = {
  // trigger
  loadingPage: PropTypes.bool.isRequired,
  needReloadHolidays: PropTypes.bool.isRequired,
  // state
  readonly: PropTypes.bool.isRequired,
  // data
  holidays: PropTypes.array.isRequired,
  fulfilmentId: PropTypes.string.isRequired,
  // redux-base
  holidaysGetRequest: PropTypes.func.isRequired,
  holidaySaveRequest: PropTypes.func.isRequired,
  holidayDeleteRequest: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(BlackoutTab);
