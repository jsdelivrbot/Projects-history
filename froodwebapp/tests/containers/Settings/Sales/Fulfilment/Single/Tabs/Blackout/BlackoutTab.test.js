import { BlackoutTab } from 'containers/Settings/Sales/Fulfilment/Single/Tabs/Blackout/BlackoutTab';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('BlackoutTab', () => {
  const props = {
    loadingPage: false,
    needReloadHolidays: false,
    readonly: false,
    holidays: [],
    fulfilmentId: '1',
    holidaysGetRequest: jest.fn(),
    holidaySaveRequest: jest.fn(),
    holidayDeleteRequest: jest.fn()
  };
  const DATE_TO_USE = new Date('2016');
  const _Date = Date;
  global.Date = jest.fn(() => DATE_TO_USE);
  global.Date.UTC = _Date.UTC;
  global.Date.parse = _Date.parse;
  global.Date.now = _Date.now;

  it('renders the BlackoutTab component', () => {
    const component = shallow(<BlackoutTab { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of BlackoutTab', () => {
    const component = shallow(<BlackoutTab { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls holidaySaveRequest in handleSave with correct param', () => {
    const holiday = {
      startDate: '1 september',
      endDate: '31 august'
    };
    const component = shallow(<BlackoutTab { ...props } />);
    component.instance().handleSave(holiday);

    expect(props.holidaySaveRequest.mock.calls.length).toBe(1);

    expect(props.holidaySaveRequest.mock.calls[0][0]).toEqual({
      id: props.fulfilmentId,
      payload: holiday
    });
  });

  it('calls holidayDeleteRequest in handleDeleteRow with correct param', () => {
    const e = {
      target: {
        id: '1'
      }
    };
    const component = shallow(<BlackoutTab { ...props } />);
    component.instance().handleDeleteRow(e);

    expect(props.holidayDeleteRequest.mock.calls.length).toBe(1);

    expect(props.holidayDeleteRequest.mock.calls[0][0]).toEqual({
      id: props.fulfilmentId,
      holidayId: e.target.id
    });
  });
});
