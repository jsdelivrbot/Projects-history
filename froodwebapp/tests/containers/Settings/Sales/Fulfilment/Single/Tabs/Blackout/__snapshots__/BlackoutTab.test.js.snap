// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlackoutTab capturing Snapshot of BlackoutTab 1`] = `
<div>
  <ReduxForm
    buttonVisible={false}
    fields={
      Array [
        Object {
          "editableForExistingItem": true,
          "header": "From",
          "name": "startDate",
          "type": "datePicker",
        },
        Object {
          "editableForExistingItem": true,
          "header": "To",
          "name": "endDate",
          "type": "datePicker",
        },
        Object {
          "editableForExistingItem": true,
          "header": "Name",
          "name": "name",
          "type": "text",
        },
      ]
    }
    handleSave={[Function]}
    handleToggleModal={[Function]}
    initialValues={
      Object {
        "endDate": "2016-01-01T00:00:00.000Z",
        "name": "",
        "startDate": "2016-01-01T00:00:00.000Z",
      }
    }
    loading={false}
    okText="Add"
    title="New Holiday"
    visible={false}
  />
  <Spin
    prefixCls="ant-spin"
    size="default"
    spinning={false}
    wrapperClassName=""
  >
    <Row
      lg={true}
      md={true}
      xs={true}
    >
      <Table
        bordered={false}
        className="table"
        columns={
          Array [
            Object {
              "dataIndex": "startDate",
              "render": [Function],
              "title": "From",
            },
            Object {
              "dataIndex": "endDate",
              "render": [Function],
              "title": "To",
            },
            Object {
              "dataIndex": "name",
              "render": [Function],
              "title": "Name",
            },
            Object {
              "dataIndex": "total",
              "render": [Function],
              "title": "Actions",
            },
          ]
        }
        dataSource={Array []}
        indentSize={20}
        loading={false}
        locale={Object {}}
        pagination={false}
        prefixCls="ant-table"
        rowKey="id"
        rowSelection={null}
        showHeader={true}
        size="small"
        useFixedHeader={false}
      />
      <NewTableRow
        addNewRowText="Add New Holiday"
        addOtherRowText="Add Another Holiday"
        hasData={false}
        onClick={[Function]}
      />
    </Row>
  </Spin>
</div>
`;
